app/models/tag_set.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class TagSet < ApplicationRecord
+  belongs_to :tag_group, class_name: 'TagGroup', optional: false
```

Comment:
Probably worth saying in a comment that, for dual index tag_group links to the 'i7' oligos, and tag2_group links to the 'i5' oligos?